### PR TITLE
Update interoperability page on docs.md

### LINKED ISSUE #1019: Update the Interoperability page
Update `/apps/concepts/interoperability/` with findings from the [`TWG 2.0 project`](https://github.com/medic/interoperability).

### PR DESCRIPTION
**Pull request by Phil-Mwago**
An update on the interoperability docs based on the Q1?TWG 2.0 work.

Closes #1019

### CODE REVIEW

File: content/en/apps/concepts/interoperability.md [hunk 1]
```diff
@@ -3,36 +3,55 @@ title: Interoperability
...
 
 Conversely to bring data in to the CHT, OpenHIM should be configured to route the updated resource to a Mediator, which then calls the relevant CHT APIs to update the document in the CHT database. This will then be replicated to users' devices as per usual.
 
-A reference application for this pattern is available in the [CHIS Interoperability repository](https://github.com/medic/chis-interoperability).
+# Important Links
+- A reference to an application repository can be accessed in the [CHT-interoperability repository](https://github.com/medic/cht-interoperability).
+- This [README](https://github.com/medic/cht-interoperability#readme) contains documentation on the CHT-interoperability project.
+- Test instances:
+- - [CHT](https://interop-cht-test.dev.medicmobile.org/medic/login?redirect=https%3A%2F%2Finterop-cht-test.dev.medicmobile.org%2F)
+- - [OpenHIM Admin Console](https://interoperability.dev.medicmobile.org/#!/login)
```
Comment (mrjones-plip) [inline content/en/apps/concepts/interoperability.md:55]: There's some repetition here, we can simplify (but admittedly add a bit of repetitiveness :shrug: )

```suggestion
-  [CHT-interoperability repository](https://github.com/medic/cht-interoperability): A reference application for the LTFU interoperability
- [CHT LTFU Test Instance](https://interop-cht-test.dev.medicmobile.org/medic/login?redirect=https%3A%2F%2Finterop-cht-test.dev.medicmobile.org%2F)
- [OpenHIM LTFU Test Instance Admin Login](https://interoperability.dev.medicmobile.org/#!/login) 
```

**Review by mrjones-plip — CHANGES_REQUESTED** [review]
Thanks for the work to clean this up and extend the info we have!

There's some pretty heavy changes I'm recommending.  Feel free to push back or grab me for an ad hoc call if you think I'm pushing in the wrong direction!

File: content/en/apps/concepts/interoperability.md [hunk 2]
```diff
@@ -3,36 +3,55 @@ title: Interoperability
...
+Interoperability refers to the ability of different systems and applications to communicate with each other and exchange data seamlessly.With interoperability, patient information can be seen, exchanged, and used across different platforms that may be disparate or relatively similar. The information/data exchanged has to be understood across the different software for these systems to become interoperable. 
 
-Interoperability is the best practice for health systems because it allows information from one system to be shared with one or more other systems with no additional development.
+# CHT Interoperability
 
-# Components
+The structure of documents in the CHT database reflect the configuration of the system, and therefore do not map directly to a FHIR message format. As the CHT is not natively FHIR standardized, we use a middleware to convert the CHT data structure into a standardized form so the other systems can read it.
 
-Useful components and reference information for interoperability include:
+{{< figure src="flow.png" link="flow.png" >}}
```
[the remark this answers is not in the document]
Comment (Phil-Mwago) [reply content/en/apps/concepts/interoperability.md:20]: This is accurate based on the arrow symbol direction and text preceding the figure. However, we have one mediator.

File: content/en/apps/concepts/interoperability.md [hunk 3]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
 linkTitle: "Interoperability"
 weight: 8
 description: >
-  Exchanging information between CHT Core and other systems
+  Two way data synchronization between the CHT Core and other health systems.
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:6]:
```suggestion
  Exchanging information between the CHT Core and other health systems 
```

I think information exchange is more accurate than synchronization in the context of interoperability.

File: content/en/apps/concepts/interoperability.md [hunk 4]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
 keywords: interoperability integrations fhir icd openhie openhim
 relatedContent: >
   apps/features/integrations
 ---
 
-# Introduction
+## Introduction 
 
-Interoperability is the ability of health information systems to work together, even if they weren't specifically designed to work together. With interoperability, patient information can be seen, exchanged, and used across different platforms. This is different from _integration_ which requires custom development to connect two specific systems together. 
+Interoperability refers to the ability of different systems and applications to communicate with each other and exchange data seamlessly.With interoperability, patient information can be seen, exchanged, and used across different platforms that may be disparate or relatively similar. The information/data exchanged has to be understood across the different software for these systems to become interoperable.
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:14]:
```suggestion
Interoperability refers to the ability of different health information systems and applications to communicate with each other and exchange data seamlessly. With interoperability, patient information can be seen, exchanged, and used across different platforms. The information/data exchanged has to be understood across the different software for these systems to become interoperable. This is different from _integration_ which requires custom development to connect two specific systems together.
```

Putting back some text from the previous version on integration, as it seems important to mention here.

File: content/en/apps/concepts/interoperability.md [hunk 5]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+This workflow describes a use case where a health facility/requesting system generates a list of patients who have missed appointments for follow-up through on the CHT. A Community Health Worker (CHW) physically visits the patients who missed the appointments and encourages them to visit the health facility. The follow-up process can also be conducted over phone or text messaging.
+In the context of the proof of concept, the intended users of the interoperable systems are Community Health Workers (CHWs) on the CHT side and Healthcare Givers on the requesting system side. System administrators can access the mediator on the administration console.
+
+The Interoperating Systems
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:38]:
```suggestion
The Interoperating Systems:
```

File: content/en/apps/concepts/interoperability.md [hunk 6]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+- [CHT](https://docs.communityhealthtoolkit.org/): Community Health Toolkit.
+- [OpenHIM](http://openhim.org/): Mediator (middleware).
+- Requesting System: [Postman](https://www.postman.com/) was used for testing purposes.
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:41]:
```suggestion
- Requesting System: For testing purposes, we used HTTP Requests.
```

I don't think it is necessary to mention Postman. Any tool for HTTP requests can work in this scenario.

File: content/en/apps/concepts/interoperability.md [hunk 7]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+
+OpenHIM was chosen as the main component of the interoperability layer and custom mediators were built to deal with the different workflows as it provides a central point of control for managing data exchange and security.
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:43]: You mention different workflows, but we only implement one for the TWG, right?

File: content/en/apps/concepts/interoperability.md [hunk 8]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+
+{{< figure workflows="LTFU Sequence Diagram.png" link="LTFU Sequence Diagram.png" >}}
 
 Conversely to bring data in to the CHT, OpenHIM should be configured to route the updated resource to a Mediator, which then calls the relevant CHT APIs to update the document in the CHT database. This will then be replicated to users' devices as per usual.
 
-A reference application for this pattern is available in the [CHIS Interoperability repository](https://github.com/medic/chis-interoperability).
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:37]: Can we keep this line and add the relevant GitHub repo? You mentioned it below, but it's worth explaining that we implemented the flow in that repo.

File: content/en/apps/concepts/interoperability.md [hunk 9]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+# Important Links
+-  [CHT-interoperability repository](https://github.com/medic/cht-interoperability): A reference application for the LTFU interoperability
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:50]:
```suggestion
- [cht-interoperability repository](https://github.com/medic/cht-interoperability): A reference application for the LTFU workflow
```

File: content/en/apps/concepts/interoperability.md [hunk 10]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+- [CHT LTFU Test Instance](https://interop-cht-test.dev.medicmobile.org/medic/login?redirect=https%3A%2F%2Finterop-cht-test.dev.medicmobile.org%2F)
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:51]:
```suggestion
- [CHT Instance with LTFU configuration](https://interop-cht-test.dev.medicmobile.org/medic/login?redirect=https%3A%2F%2Finterop-cht-test.dev.medicmobile.org%2F)
```

File: content/en/apps/concepts/interoperability.md [hunk 11]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+- [OpenHIM LTFU Test Instance Admin Login](https://interoperability.dev.medicmobile.org/#!/login)
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:52]:
```suggestion
- [OpenHIM Admin Console](https://interoperability.dev.medicmobile.org/#!/login) 
```

File: content/en/apps/concepts/interoperability.md [hunk 12]
```diff
@@ -3,36 +3,52 @@ title: Interoperability
...
+{{< figure workflows="LTFU Sequence Diagram.png" link="LTFU Sequence Diagram.png" >}}
 
 Conversely to bring data in to the CHT, OpenHIM should be configured to route the updated resource to a Mediator, which then calls the relevant CHT APIs to update the document in the CHT database. This will then be replicated to users' devices as per usual.
 
-A reference application for this pattern is available in the [CHIS Interoperability repository](https://github.com/medic/chis-interoperability).
+# Important Links
+-  [CHT-interoperability repository](https://github.com/medic/cht-interoperability): A reference application for the LTFU interoperability
+- [CHT LTFU Test Instance](https://interop-cht-test.dev.medicmobile.org/medic/login?redirect=https%3A%2F%2Finterop-cht-test.dev.medicmobile.org%2F)
+- [OpenHIM LTFU Test Instance Admin Login](https://interoperability.dev.medicmobile.org/#!/login) 
+
```
Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:53]:
```suggestion
```

Comment (andrablaj) [inline content/en/apps/concepts/interoperability.md:53]: Can we add a link to the mediator instance?

**Review by andrablaj — CHANGES_REQUESTED** [review]
Thanks @Phil-Mwago!

I added a few suggestions. I can't see the sequence diagram when I run the docs locally. Any idea why? Shouldn't the `.png` file be in this PR?

File: content/en/apps/concepts/interoperability.md [hunk 13]
```diff
@@ -3,36 +3,55 @@ title: Interoperability
...
 ---
 
-# Introduction
+# Introduction 
 
-Interoperability is the ability of health information systems to work together, even if they weren't specifically designed to work together. With interoperability, patient information can be seen, exchanged, and used across different platforms. This is different from _integration_ which requires custom development to connect two specific systems together. 
+Interoperability refers to the ability of different systems and applications to communicate with each other and exchange data seamlessly.With interoperability, patient information can be seen, exchanged, and used across different platforms that may be disparate or relatively similar. The information/data exchanged has to be understood across the different software for these systems to become interoperable. 
 
-Interoperability is the best practice for health systems because it allows information from one system to be shared with one or more other systems with no additional development.
+# CHT Interoperability
```
[the remark this answers is not in the document]
Comment (Phil-Mwago) [reply content/en/apps/concepts/interoperability.md]: I wanted uniformity with the blog posts we have put out there but on further review, we can use the original text.

File: content/en/apps/concepts/interoperability.md (hunk 7 above)
Comment (Phil-Mwago) [reply content/en/apps/concepts/interoperability.md:42]: Yes, we only implement the LTFU.

File: content/en/apps/concepts/interoperability.md (hunk 12 above)
Comment (Phil-Mwago) [reply content/en/apps/concepts/interoperability.md:48]: Done

File: content/en/apps/concepts/interoperability.md (hunk 8 above)
Comment (Phil-Mwago) [reply content/en/apps/concepts/interoperability.md:37]: Alright, adding the cht-instrop repo.

File: content/en/apps/concepts/interoperability.md [hunk 14]
```diff
@@ -3,36 +3,55 @@ title: Interoperability
...
 
-Useful components and reference information for interoperability include:
+{{< figure src="flow.png" link="flow.png" >}}
 
-- [OpenHIE](https://ohie.org/) defines the architecture for an interoperability layer.
-- [OpenHIM](http://openhim.org/) is a middleware component designed to ease interoperability between systems.
-- [HL7 FHIR](https://www.hl7.org/fhir/index.html) is a messaging format to allow all systems to understand the format of the message.
-- [ICD-11](https://www.who.int/standards/classifications/classification-of-diseases) and [LOINC](https://loinc.org/) are examples of classification systems.
+Interoperability is the best practice for health systems because it allows information from one system to be shared with one or more other systems with no additional development. Interoperability allows technical teams to scale in an efficient and repeatable manner due to the already predefined standards.
```
[the remark this answers is not in the document]
Comment (Phil-Mwago) [reply content/en/apps/concepts/interoperability.md]: Done

File: content/en/apps/concepts/interoperability.md [hunk 15]
```diff
@@ -3,36 +3,40 @@ title: Interoperability
...
+## CHT Interoperability
 
-Useful components and reference information for interoperability include:
+The native CHT database structure does not map directly to a [Fast Healthcare Interoperability Resources (FHIR)](http://www.hl7.org/fhir/) message format. To be compatible, we use a middleware to convert the CHT data structure into a standardized JSON format so the other systems can read it.
 
-- [OpenHIE](https://ohie.org/) defines the architecture for an interoperability layer.
-- [OpenHIM](http://openhim.org/) is a middleware component designed to ease interoperability between systems.
-- [HL7 FHIR](https://www.hl7.org/fhir/index.html) is a messaging format to allow all systems to understand the format of the message.
-- [ICD-11](https://www.who.int/standards/classifications/classification-of-diseases) and [LOINC](https://loinc.org/) are examples of classification systems.
+{{< figure src="flow.png" link="flow.png" >}}
```
Comment (mrjones-plip) [inline content/en/apps/concepts/interoperability.md:22]: Let's use the new mermaid chart Josh added!

I'd make this a suggestion, but GH isn't playing nice with me having a code block inside my suggestion block :shrug: 

Here's the code to uses , but the first line needs to wrapped in open ` ```mermaid` and close  ` ``` ` :

```
graph LR
cht[CHT]
mediator_a([Mediator])
mediator_b([Mediator])
openhim[OpenHIM]
cht -- Outbound push\nfa:fa-arrow-right --- mediator_a
cht -- API request\nfa:fa-arrow-left --- mediator_b
mediator_a -- Request\nfa:fa-arrow-right --- openhim
mediator_b -- Channel\nfa:fa-arrow-left --- openhim
```

**Review by mrjones-plip — CHANGES_REQUESTED** [review]
looking good!

let's:
* replace `flow.png` with mermaid chart
* delete `flow.png` and `LTFU Sequence Diagram.png`

File: content/en/apps/concepts/interoperability.md (hunk 15 above)
Comment (Phil-Mwago) [reply content/en/apps/concepts/interoperability.md:22]: Thanks, adding the mermaid.

**Comment by Phil-Mwago** [comment]
> Requested changes complete.

**Review by mrjones-plip — APPROVED** [review]
lgtm!

we remove the other image in #1090  - so this one is good as is - congrats and thanks for working with me on this!

File: content/en/apps/concepts/interoperability/flow.png
Comment (Phil-Mwago) [inline content/en/apps/concepts/interoperability/flow.png:1]: Resolved... The LTFU Sequence diagram and text will go to the examples page.

**Comment by Phil-Mwago** [comment]
> Thanks @Phil-Mwago!
> 
> I added a few suggestions. I can't see the sequence diagram when I run the docs locally. Any idea why? Shouldn't the `.png` file be in this PR?

We will add the diagram in the examples page.

**Comment by Phil-Mwago** [comment]
> Thanks @Phil-Mwago!
> 
> I added a few suggestions. I can't see the sequence diagram when I run the docs locally. Any idea why? Shouldn't the `.png` file be in this PR?

We will add the digram in the examples page.

**Review by samuelimoisili — APPROVED** [review]

**Pull request merged**